client/lib/user/index.js

```diff
@@ -17,14 +11,3 @@ export default function () {
 	}
 	return _user;
```

Comment:
Now that we've removed the last Flux bit from here, this does look a bit awkward. I wonder if we can remove the indirection and directly lead to `/user` instead of instantiating this one. Not necessarily in this PR, of course.

Reply:
I think we could remove the whole "user factory" convention:
```js
import userFactory from 'calypso/lib/user';
const user = userFactory();
```
and replace it with a direct import of a singleton object:
```js
import user from 'calypso/lib/user';
```

The `calypso/lib/user` module used to have side-effects: merely creating the object with `userFactory()` started fetching the user info from network. But that's long gone and now we do an explicit `user.fetch()` everywhere it's needed.

Reply:
Sounds good to me, and would be a good follow-up after this one is merged.